### DISCUSSION TITLE
Listing flux suspended resources

### DISCUSSION THREAD

**Discussion by gberche-orange** [fluxcd/flux2#2494] [Q&A]
Whats the best option to list all suspended resources ? 

The [flux get](https://fluxcd.io/docs/cmd/flux_get/) supports a `--status-selector` which does not cover the spec.suspend condition.

One option is to post-process the output to filter on suspended=false
* in the `flux get all` output
* in raw kubectl output: `$ kubectl get --all-namespaces  kustomizations.kustomize.toolkit.fluxcd.io -o json | jq -r '.items[] | select(.spec.suspend==true) | [.metadata.name,.spec.suspend] | @tsv'`

Would be great to have a built-in way to select suspended resources.

**Comment by kingdonb** [comment]
I don't think we have this feature yet, but it is a good idea! There are some PRs recently related to printer interface, I am not sure if that will be helpful or on the way;

But for example, if there was a way to call `flux resume` so it only pings suspended kustomizations would be one example use case.

**Comment by jflambert** [comment]
Ended up here from googling "flux get suspended list" so yes, would love that functionality.

**Comment by fragolinux** [comment]
@kingdonb hi! Just added this as k9s plugins, and just made a PR to have them included... till k9s allows us to add custom columns based on .spec in their views... https://github.com/derailed/k9s/pull/1584
using this plugin, together with the others I submitted months ago and now in default k9s repo, you can press:
- shift-s to show suspended helmreleases and kustomization
- shift-t to toggle suspend/resume a specific resource
- shift-r to force a reconcile of the selected resource...

**Comment by makkes** [comment]
While I agree that having such a selector in the `get` or the `resume` command, `--status-selector` wouldn't be the right place to put it because that is specifically targeted at resource conditions while the `suspend` field is part of the spec. So maybe a dedicated flag `--suspended=[true|false]` (or `suspend-only` (defaulting to `false`) would make more sense here.

**Comment by jflambert** [comment]
Should this discussion be turned into an issue? I don't suppose any work was done on this?

Meanwhile, the following works well enough (as long as the output doesn't change in future flux versions!)

`flux get kustomizations | awk '{ if ( $3 == "True" ) { printf "* %s\n", $1 } }'`
